presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java

```diff
@@ -396,6 +423,8 @@ public Object getJavaValue(Object nativeValue)
             case BOOLEAN:
             case DOUBLE:
             case COUNTER:
+            case SMALLINT:
+            case TINYINT:
```

Comment:
why SMALLINT, TINYINT is treated differently than INT?
i don't know what this function is supposed to do, but this looks suspicious.